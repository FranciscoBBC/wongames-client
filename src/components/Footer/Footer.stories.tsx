import Footer from '.'
import { Story, Meta } from '@storybook/react'

export default {
  title: 'Footer',
  component: Footer
} as Meta

export const basic: Story = () => (
  <div style={{ maxWidth: '130rem', margin: '0 auto' }}>
    <Footer />
  </div>
)
