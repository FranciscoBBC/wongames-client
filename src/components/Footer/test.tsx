import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Footer from '.'

describe('<Footer />', () => {
  it('should render footer links', () => {
    const { container } = renderWithTheme(<Footer />)

    expect(
      screen.getByRole('heading', { name: /contact/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /follow us/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /links/i })).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /location/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the mobile footer', () => {
    renderWithTheme(<Footer />)
    expect(screen.getByTestId('content-wrapper')).toHaveStyle({
      'grid-template-columns': 'repeat(2,1fr)'
    })
  })
  it('should render the desktop footer', () => {
    renderWithTheme(<Footer />)
    expect(screen.getByTestId('content-wrapper')).toHaveStyleRule(
      'grid-template-columns',
      'repeat(4,1fr)',
      {
        media: '(min-width: 768px)'
      }
    )
  })
})
