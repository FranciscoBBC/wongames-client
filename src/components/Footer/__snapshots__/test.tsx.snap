// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render footer links 1`] = ` `;
